google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/ChannelCache.java

```diff
@@ -43,17 +53,34 @@ public ChannelCache() {
   void onShutdown() {
     channels.forEachValue(
         channels.size(),
-        channel -> {
+        channels -> {
           try {
-            channel.shutdownNow().awaitTermination(60, TimeUnit.SECONDS);
+            for (ManagedChannel channel : channels) {
+              channel.shutdownNow().awaitTermination(60, TimeUnit.SECONDS);
+            }
           } catch (InterruptedException e) {
             e.printStackTrace();
           }
         });
   }
 
   public Channel get(String target) {
-    return channels.computeIfAbsent(target, channelFactory);
+    List<ManagedChannel> channelList = channels.computeIfAbsent(target, this::newChannels);
+    return channelList.get(rand.nextInt(channelList.size()));
+  }
+
+  private List<ManagedChannel> newChannels(String target) {
+    int numberOfChannels = NUMBER_OF_CHANNELS_PER_TARGET;
+    String numberOfChannelsOverride = System.getProperty(NUMBER_OF_CHANNELS_PER_TARGET_VM_OVERRIDE);
+    if (numberOfChannelsOverride != null && !numberOfChannelsOverride.isEmpty()) {
+      numberOfChannels = Integer.parseInt((numberOfChannelsOverride));
```

Comment:
This can throw NumberFormatException into a weird place in the code. Catch this and log at severe if this happens.

Reply:
Done